### PR TITLE
50 teibot test

### PR DESCRIPTION
**Pull request by ths2o**

### CODE REVIEW

File: playground/tei_bot_v2_util.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+)
+
+
+@dataclass
+class Position1D:
+    connected_size: int
+    position: List[int]
+
+
+def get_white(mat) -> np.array:
```
Comment (chzlithoo) [inline playground/tei_bot_v2_util.py:38]: please add type of `mat`

File: playground/tei_bot_v2_util.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    mat = np.where(mat != 2, 0, mat)
+    mat = np.where(mat == 2, 1, mat)
+    return mat
+
+
+def get_black(mat) -> np.array:
```
Comment (chzlithoo) [inline playground/tei_bot_v2_util.py:44]: please add type of `mat`

File: playground/tei_bot_v2_util.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    connected_positions_max_by_size = {}
+    for i in range(1, 7):
+        ll = list(filter(lambda x: x[0] == i, max_connected_positions))
+        ll2 = list(map(lambda x: (int(x[1][0:2]), int(x[1][2:4])), ll))
+        connected_positions_max_by_size[i] = ll2
+
+    return connected_positions_max_by_size
+
+
+def choose_connected_position(dic, n_connected) -> np.array:
```
Comment (chzlithoo) [inline playground/tei_bot_v2_util.py:163]: please add type of `dic` and `n_connected`

File: playground/tei_bot_v2_util.py [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+    xx1 = dic[n_connected]
+    xx2 = (
+        np.array(list(map(lambda x: x[0], xx1)), dtype="int64"), np.array(list(map(lambda x: x[1], xx1)), dtype="int64")
+    )
+    put_array[xx2] = 1
+
+    return put_array
+
+
+def neighbor_sum(mat) -> np.array:
```
Comment (chzlithoo) [inline playground/tei_bot_v2_util.py:174]: please add type of `mat`

File: playground/tei_bot_v2_util.py [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+    s = 2
+    kernel = np.ones(s * s, dtype="int32").reshape((s, s))
+    c = convolve(mat, kernel, mode='constant')
+    c = c / (s * s)
+    return c
+
+
+def suggest_position_by_connected_element(mat, turn) -> np.array:
```
Comment (chzlithoo) [inline playground/tei_bot_v2_util.py:182]: please add type of `mat` and `turn`

File: playground/tei_bot_v2_util.py [hunk 6]
```diff
@@ -0,0 +1,269 @@
+import numpy as np
+from operator import itemgetter
+from itertools import groupby
+from random import choice
+from dataclasses import dataclass
+from scipy.ndimage import convolve
+from typing import List, Tuple, Dict
+
+a = np.array(
```
Comment (hyoungseok) [inline playground/tei_bot_v2_util.py:9]: seems no usage

File: playground/bot.py [hunk 7]
```diff
@@ -1,6 +1,7 @@
 from typing import List
 from random import randint
 from board import Stone, BoardConfig
+from tei_bot_v2 import TeiBot as TeiBot_v2
```
Comment (hyoungseok) [inline playground/bot.py:4]: seems un-natural.
better to change class name rather than using "as"

**Comment by chzlithoo** [comment]
@ths2o please change `tei` into `tei_v2` in https://github.com/allmeat/connect6/blob/master/repo/collect_log.py#L51

if you cannot see this file in your branch

```bash
git checkout 50-teibot_test
git pull origin master
```
and then 
```bash
git push origin 50-teibot_test
```

**Pull request merged**